### DISCUSSION TITLE
ipv6 可以解析出地址了，却ping不通(已解决，非常感谢作者)

### DISCUSSION THREAD

**Discussion by Leno709394** [kkkgo/PaoPaoDNS#32] [问答和帮助]
C:\Users\Administrator>nslookup 6.ipw.cn 192.168.168.252
服务器:  UnKnown
Address:  192.168.168.252

名称:    6.ipw.cn
Address:  2402:4e00:1013:e500:0:9671:f018:4947

C:\Users\Administrator>ping 6.ipw.cn

正在 Ping 6.ipw.cn [2402:4e00:1013:e500:0:9671:f018:4947] 具有 32 字节的数据:
请求超时。
请求超时。
请求超时。
请求超时。

2402:4e00:1013:e500:0:9671:f018:4947 的 Ping 统计信息:
    数据包: 已发送 = 4，已接收 = 0，丢失 = 4 (100% 丢失)，

192.168.168.252 是paopaodns ip
折腾半天，可以解析v6地址了。却ping不通，又是啥问题

**Comment by kkkgo** [comment]
能解析出ipv6地址说明DNS工作正常。你需要检查你的IPV6分配的地址是否已经过期，分配的前缀是否和网关分配的一致，以及你路由器的ipv6路由是否存在问题，总之，很多家宽获取ipv6地址都存在过期没自动刷新地址的问题，这需要主路由的设置来解决，DNS这一步已经完成任务了。

**Reply by Leno709394** [reply]
搞定了。paopaodns+paopao网关 完美实现了我的需求。 非常感谢作者大大
实现了泡泡说明里的。国内域名有v6的走v6.。 国外只返回v4
<img src="https://s1.imagehub.cc/images/2023/07/21/1efc842c617a84b56.jpeg" alt="1efc842c617a84b56.jpeg" border="0" />

**Reply by kkkgo** [reply]
家宽v6地址存在地址过期的问题，当地址过期了又没有及时刷新的情况下，设备优先走v6可能会造成上网卡顿的感觉。如果设置成`IPV6=only6`的话就只对纯v6网站解析ipv6地址（比如`6.ipw.cn`）。总之，IPV6的几个选项（`no`,`yes`,`only6`,`raw`）是看个人需求设置的，根据自己场景选择就好。

**Reply by Leno709394** [reply]
感谢给出意见。我也刚发现解析v6 没 v4 快。。我搞only6 试试

**Reply by Geilberton** [reply]
求个相关配置，同样的ppgw ppdns，局域网内不需要海淘的设备v6是正常的，但是需要海淘的设备dns转发到ppdns后却是不行的，换个支持v6的节点才行。
想达到楼主同样的方案效果，国内AAAA，国外只返回A（避免更换不支持的节点）。
感谢~

**Reply by kkkgo** [reply]
IPV6环境变量设置成yes就行。

**Reply by Geilberton** [reply]
就是设置了都不行的......不过是我后期改的而不是布置docker时设置的。

**Reply by kkkgo** [reply]
docker启动后改不了的吧，docker compose可以改。

**Reply by Geilberton** [reply]
docker stop后改的，行吧，我试试重新部署看看行不行~

**Reply by Geilberton** [reply]
非常感谢，之前暂停更改并没有生效。
重新部署后可以了。看来还是装个compose方便点，不然每次更改变量都要重新部署。
感谢感谢~

**Reply by kkkgo** [reply]
新版本添加了custom_env功能，改环境变量不需要重新部署了。详情参考更新日志。https://github.com/kkkgo/PaoPaoDNS/discussions/61
